Assets/MixedRealityToolkit.Services/InputSystem/FocusProvider.cs

```diff
@@ -704,19 +715,23 @@ private void UpdatePointer(PointerData pointer)
                     // Otherwise, continue
                     LayerMask[] prioritizedLayerMasks = (pointer.Pointer.PrioritizedLayerMasksOverride ?? FocusLayerMasks);
 
-                    // Clear the hit
-                    pointer.ClearHits();
-
                     // Perform raycast to determine focused object
-                    QueryScene(pointer, prioritizedLayerMasks);
+                    PointerHitResult hit = new PointerHitResult(); // TODO: cache and clear
```

Comment:
>// TODO: cache and clear [](start = 67, length = 24)

Nit: Please clean up the TODO prior to checking this in (i.e. shouldn't be per-update allocations)

Reply:
(Same with the thing down below) - i.e. hitUi

---
In reply to: [270985314](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/3749#discussion_r270985314) [](ancestors = 270985314)

Reply:
Thanks for pointing this out. I actually forgot to address this